src/lib.rs

```diff
@@ -20,49 +52,93 @@ impl Cleverbot {
     /// Creates a new Cleverbot instance.
     /// * `user` - The API User.
     /// * `key` - The API Key.
-    pub fn new(user: String, key: String, nick: Option<String>) -> Result<Cleverbot, String> {
+    pub fn new(user: String, key: String, nick: Option<String>) -> Result<Cleverbot, CleverbotError> {
         let mut response = {
             let mut args = vec![
                 ("user", &*user),
                 ("key", &*key),
             ];
             if let Some(ref nick) = nick { args.push(("nick", &*nick)) }
-            request("https://cleverbot.io/1.0/create", &*args)
+            match request("https://cleverbot.io/1.0/create", &*args) {
+                Ok(response) => response,
+                Err(err) => return Err(CleverbotError::Io(err)),
+            }
         };
         let mut body = String::new();
-        response.read_to_string(&mut body).unwrap();
-        let json: BTreeMap<String, String> = serde_json::from_str(&body).unwrap();
-        let result = json.get("status").unwrap();
-        if result == "success" {
-            Ok(Cleverbot {
-                user: user,
-                key: key,
-                nick: json.get("nick").unwrap().to_string(),
-            })
-        } else {
-            Err(result.to_string())
+        try!(response.read_to_string(&mut body));
+        let json: BTreeMap<String, String> = try!(serde_json::from_str(&body));
+        let opt_result = json.get("status");
+        let result = {
```

Comment:
No need to have this extra pair of braces since the interior is just a block expression anyway.